src/modules/SX126x/STM32WLx_Module.h

```diff
@@ -0,0 +1,52 @@
+/*
+
+Copyright (c) 2022 STMicroelectronics
+
+This file is licensed under the MIT License: https://opensource.org/licenses/MIT
+*/
+
+#if !defined(_RADIOLIB_STM32WLX_MODULE_H)
+#define _RADIOLIB_STM32WLX_MODULE_H
+
+#include "../../TypeDef.h"
+
+#if !defined(RADIOLIB_EXCLUDE_STM32WLX)
+
+#include "../../Module.h"
+
+/*!
+ * \class STM32WLx_Module
+ *
+ * This is a subclass of Module to be used with the STM32WLx driver.
+ *
+ * It is used to override some callbacks, allowing access to some of the
+ * radio control signals that are wired to internal registers instead of
+ * actual GPIO pins.
+ */
+class STM32WLx_Module : public Module {
+  // Note: We cannot easily override any methods here, since most calls
+  // are non-virtual and made through a Module*, so they would not be
+  // calling any overridden methods. This means this class works by
+  // overriding some of the callbacks in its constructor.
+
+  public:
+    STM32WLx_Module();
+
+#if !defined(RADIOLIB_GODMODE)
+  private:
+#endif
+
+    // TODO: Do we need copy constructor or assignment operator? Or mabye
+    // delete them? Module has defined them, but where are they used?
```

Comment:
They are not strictly necessary and are not used in the library itself or in the examples. However, if the user needs them for whatever reason, they are provided for completeness sake.

Reply:
Ok, makes sense to have them. However, I see that if you do not define any, C++ will automatically define a [copy constructor]() and [copy assignment operator](https://en.cppreference.com/w/cpp/language/copy_assignment) (unless they are deleted, but I don't think that happens here), so I think I can just remove the comment and it will be ok.

However, looking at the copy assignment operator for Module, I see that it only copies a subset of the fields. I'm not sure if that's intentional, but I doubt that the copy assignment is at all useful right now. In particular:
 - None of the callbacks are copied. It seems these fields were added in 8111bc058fe5818082c347370a7af00bf1ff286f, but they were not added to the copy assignment constructor.
 - The `_spi`, `_spiSettings` and `_initInterface` fields are not copied. The former two were originally copied, but this was removed for some reason in 8111bc058fe5818082c347370a7af00bf1ff286f. The latter was I think never copied (but didn't look closely).
 - The `_rfSwitchPins` and `_rfSwitchTable` fields are not copied. I hadn't considered adding them, because the `_rxEn` and `_txEn` fields they replaced weren't copied either (it seems these fields were added after the copy operator was added).

I actually think that to be useful, these copy operators should really just copy *all* fields, right? Or are there fields that should be omitted? If not, I think they can just be omitted, relying on the compiler to generate default versions that just copy all fields . In C++11 you can declare them with `= default` to force generating the defaults, or (I think) throw an error if defaults are not possible (e.g. with reference or const members), might be useful to add that (in appropriate C++ preprocessor guards).

Reply:
You're right, the copy operators should copy all fields. I didn't know compiler generates them by default (though it does make sense actually, come to think about it).

In the scope of this PR, you can just remove the comment. I will also remove the copy and assignment constructors from Module class (since it seems that the reason they only copy some fields is that I simply forgot to add them to the constructors ... compiler should be smarter than human in this case).

Reply:
Resolved in 02bd4293b7418210931bd40c543c4556e7984fcb.